### PR TITLE
Fix : allow some url to be allowed in `script-src` CSP

### LINKED ISSUE #651: CSP cause Netlify CMS failed to open
## Description

On #634 we have CSP related pull-request for issue #611

It cause our CMS Netlify failed to open on https://www.wargabantuwarga.com/admin/

Preview Blank Page:

Screen Shot 2021-08-12 at 14 39 54

Error:

Screen Shot 2021-08-12 at 14 39 48

```
Content Security Policy: The page’s settings blocked the loading of a resource at https://identity.netlify.com/v1/netlify-identity-widget.js (“script-src”).
Content Security Policy: The page’s settings blocked the loading of a resource at https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js (“script-src”).
```

## Expected Behaviour

Should not effected

## Additional Info

Our homepage also show error

Screen Shot 2021-08-12 at 14 43 59

```
Content Security Policy: The page’s settings blocked the loading of a resource at https://analytics.google.com/g/collect?v=2&tid=G-Q32QC5END5&gtm=2oe8b0&_p=1768343212&sr=1920x1080&_gaz=1&ul=en-us&cid=1694819148.1628219111&_s=1&dl=https%3A%2F%2Fwww.wargabantuwarga.com%2F&dt=Informasi%20Faskes%20%26%20Alkes%20untuk%20COVID-19%20%7C%20Warga%20Bantu%20Warga&sid=1628753583&sct=9&seg=1&en=page_view&ep.debug_mode=true (“default-src”).
```

### PR DESCRIPTION
**Pull request by asaadam**
Closes #651 

## Description

add this url, to be allowed in `script-src` CSP
- *.netlify.com 
- unpkg.com 
- analytics.google.com 
- *.google-analytics.com

### CODE REVIEW

**Review by mazipan — APPROVED** [review]
Looks good, let's wait to check on the Netlify Preview first ya

**Comment by asaadam** [comment]
image
after see the preview
'unsafe-eval' must be allowed.

it's this okay?

**Comment by mazipan** [comment]
Can we pass different config for different domain?

**Comment by asaadam** [comment]
> Can we pass different config for different domain?
> 
> The error on the Homepage are Gone for now.
> But for the CMS still exist.

right now I don't know how we can pass different configs for different domains.

but for now, if we want to quick fix, I will allow `'unsafe-eval'`
then we will figure out later

**Comment by mazipan** [comment]
```
  for = "/admin*"
  [headers.values]
    // Set different value
  for = "/*"
  [headers.values]
    X-Frame-Options = "DENY"
    X-XSS-Protection = "1; mode=block"
    X-Content-Type-Options = "nosniff"
    Referrer-Policy = "same-origin"
    Content-Security-Policy = "default-src 'self'; style-src 'self' 'unsafe-inline'; script-src 'self' 'unsafe-inline' *.googletagmanager.com *.netlify.com unpkg.com analytics.google.com *.google-analytics.com; img-src 'self' blob: data: https:; frame-ancestors 'none'; "
    Permissions-Policy = "camera=(), microphone=(), geolocation=(), interest-cohort=()"
```

**Comment by asaadam** [comment]
> How about differentiate based on path?
> 
> ```
>   for = "/admin*"
>   [headers.values]
>     // Set different value
>   for = "/*"
>   [headers.values]
>     X-Frame-Options = "DENY"
>     X-XSS-Protection = "1; mode=block"
>     X-Content-Type-Options = "nosniff"
>     Referrer-Policy = "same-origin"
>     Content-Security-Policy = "default-src 'self'; style-src 'self' 'unsafe-inline'; script-src 'self' 'unsafe-inline' *.googletagmanager.com *.netlify.com unpkg.com analytics.google.com *.google-analytics.com; img-src 'self' blob: data: https:; frame-ancestors 'none'; "
>     Permissions-Policy = "camera=(), microphone=(), geolocation=(), interest-cohort=()"
> ```

ah I see, let's give it a shot

File: netlify.toml [hunk 1]
```diff
@@ -4,15 +4,27 @@
 
 # Temporary header config until Netlify supports setting `headers` on `next.config.js`
 # https://github.com/netlify/netlify-plugin-nextjs/issues/150
+
+[[headers]]
+for = "/admin*"
+  [headers.values]
+     X-Frame-Options = "DENY"
```
Comment (mazipan) [inline netlify.toml:11]: i see some extra space here?

Comment (asaadam) [reply netlify.toml]: fixed

File: netlify.toml [hunk 2]
```diff
@@ -4,15 +4,27 @@
...
+
 [[headers]]
   for = "/*"
   [headers.values]
     X-Frame-Options = "DENY"
     X-XSS-Protection = "1; mode=block"
     X-Content-Type-Options = "nosniff"
     Referrer-Policy = "same-origin"
-    Content-Security-Policy = "default-src 'self'; style-src 'self' 'unsafe-inline'; script-src 'self' 'unsafe-inline' *.googletagmanager.com; img-src 'self' blob: data: https:; frame-ancestors 'none'; "
+    Content-Security-Policy = "default-src 'self'; style-src 'self' 'unsafe-inline'; script-src 'self' 'unsafe-inline' *.googletagmanager.com *.netlify.com unpkg.com analytics.google.com *.google-analytics.com; img-src 'self' blob: data: https:; frame-ancestors 'none'; "
```
Comment (mazipan) [inline netlify.toml:25]: I think here, we don't need to allow `*.netlify.com unpkg.com`

Comment (mazipan) [reply netlify.toml:25]: But it can be work later

Comment (mazipan) [reply netlify.toml:25]: Let's see the result first.

Comment (asaadam) [reply netlify.toml:25]: ah yap, since it's only needed in admin page

File: next.config.js [hunk 3]
```diff
@@ -4,7 +4,7 @@ const withBundleAnalyzer = require("@next/bundle-analyzer")({
 
 const ContentSecurityPolicy = `
   default-src 'self';
-  script-src 'self' 'unsafe-inline' *.googletagmanager.com 'unsafe-eval';
+  script-src 'self' 'unsafe-inline' 'unsafe-eval' *.googletagmanager.com *.netlify.com unpkg.com analytics.google.com *.google-analytics.com;
```
Comment (mazipan) [inline next.config.js:7]: we also need to align `next.config.js`, lets work on seperate PR

File: netlify.toml (hunk 2 above)
Comment (mazipan) [reply netlify.toml:25]: Let's keep this to be work on the next PR

File: next.config.js (hunk 3 above)
Comment (asaadam) [reply next.config.js:7]: okay, 
right now, is the admin page accessible?

**Comment by mazipan** [comment]
Still not working, so let's try to allow on all pages for now @asaadam

**Comment by asaadam** [comment]
> Still not working, so let's try to allow on all pages for now @asaadam

okay, working on it

**Comment by mazipan** [comment]
Works in my end.
let's discuss later with @resir014 ya to tackle this issue properly.

**Comment by mazipan** [comment]
@all-contributors please add @asaadam for code

**Pull request merged**